robot-code/micro-python/sumobot.py

```diff
@@ -110,4 +108,31 @@ def drive(self, speed):
             self.pin2.duty(duty_cycle)
         else:
             self.pin1.duty(0)
-            self.pin2.duty(0)
\ No newline at end of file
+
+class Sensor:
+    ### Sets up a sensor given pin number and sensor type
+    ### Does not return
+    def __init__(self, pin_number, pin_type):
+        assert pin_type=="analog" or pin_type=="digital"
+        self.pin_num = pin_number
+        if pin_type=="analog":
+            self.pin_type = "analog"
+            self.reading = ADC(Pin(self.pin_num))
+        else:
+            self.pin_type = "digital"
+            self.reading = Pin(self.pin_num, Pin.IN)
+
+    ### Reads analog signals coming from the sensor
+    ### Returns analog value
+    def read_analog(self):
```

Comment:
You can reconfigure on the fly. Creating two different functions that can that can cause a hard crash if the user calls them incorrectly might not be the best approach. Why not just reconfigure the pin and return the appropriate type instead?